### PR TITLE
Add Field Medical History for Patient

### LINKED ISSUE #23: As a GP clinic management staff, I can access a patient's medical history
...so that I can I can provide more informed treatment and care to the patient

### PR DESCRIPTION
**Pull request by angkyakdifp**
Resolves #23

### CODE REVIEW

File: src/main/java/seedu/address/model/person/MedicalHistory.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+/**
+ * Represents a Patient's medical history in the address book.
+ */
+public class MedicalHistory {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Medical History should only contain alphanumeric characters and spaces, and it should not be blank";
+
+    /*
+     * The first character of the Specialty must not be a whitespace,
```
Comment (alvinlim277) [inline src/main/java/seedu/address/model/person/MedicalHistory.java:14]: I think you meant to write Medical History here.

File: src/main/java/seedu/address/model/person/MedicalHistory.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+     * @param medicalHistoryInput A valid specialty.
+     */
+    public MedicalHistory(String medicalHistoryInput) {
+        requireNonNull(medicalHistoryInput);
+        checkArgument(isValidMedicalHistory(medicalHistoryInput), MESSAGE_CONSTRAINTS);
+        this.value = medicalHistoryInput;
+    }
+
+    /**
+     * Returns true if a given string is a valid specialty.
```
Comment (alvinlim277) [inline src/main/java/seedu/address/model/person/MedicalHistory.java:33]: Same as above.

File: src/test/java/seedu/address/model/person/MedicalHistoryTest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    @Test
+    public void isValidMedicalHistory() {
+        // null specialty
+        assertThrows(NullPointerException.class, () -> MedicalHistory.isValidMedicalHistory(null));
+
+        // invalid specialties
+        assertFalse(MedicalHistory.isValidMedicalHistory("")); // empty string
+        assertFalse(MedicalHistory.isValidMedicalHistory(" ")); // spaces only
+        // valid specialties
+        assertTrue(MedicalHistory.isValidMedicalHistory(VALID_MEDICAL_HISTORY_OSTEOPOROSIS));
```
Comment (alvinlim277) [inline src/test/java/seedu/address/model/person/MedicalHistoryTest.java:32]: I think you meant to write medical history in these comments as well.

File: src/test/java/seedu/address/testutil/PatientBuilder.java [hunk 4]
```diff
@@ -1,22 +1,49 @@
...
 
 /**
  * A utility class to help with building Patient objects.
  */
 public class PatientBuilder extends PersonBuilder {
+    public static final String DEFAULT_MEDICAL_HISTORY = "Anemia";
+    private MedicalHistory medicalHistory;
+
+    /**
+     * Constructor that assigns a default specialty.
```
Comment (alvinlim277) [inline src/test/java/seedu/address/testutil/PatientBuilder.java:14]: I think you meant to write medical history instead of specialty here as well.

**Review by alvinlim277 — COMMENTED** [review]
Overall LGTM, the implementation of patients' medical history is well done! Just some comments here and there that reference the wrong class. Additionally, codecov checks are failing so more tests might need to be added before the merge.

**Review by vansh284 — APPROVED** [review]
LGTM. Would be nice if the code coverage could be improved in the future.

**Pull request merged**